apps/hash-graph/lib/graph/src/snapshot.rs

```diff
@@ -122,10 +181,14 @@ where
     )]
     pub async fn restore_snapshot(
         &mut self,
-        snapshot: impl Stream<Item = SnapshotEntry> + Send,
+        snapshot: impl Stream<Item = Result<SnapshotEntry, impl Context>> + Send,
     ) -> Result<(), SnapshotRestoreError> {
         let mut snapshot = pin!(snapshot);
         while let Some(entry) = snapshot.next().await {
+            let entry = entry
+                .change_context(SnapshotRestoreError::Canceled)
+                .attach_printable("reading the records resulted in an error")?;
```

Comment:
This doesn't seem like the most helpful message, is there even a point adding this if we're just using a `Canceled` error

Reply:
This error will happen when the input stream (i.e. the records, which are being read) is an `Err` variant. Ideally, we don't pass in `Result` at all, this was discussed in https://github.com/hashintel/hash/pull/2357#discussion_r1157198072. I most likely go with a cancelation token but this is out of the scope of this PR as I didn't make the discoveries for that, yet.

Reply:
Fair enough, I just meant that the display/debug impl of `SnapshotRestoreError::Canceled` basically says the same as `"reading the records resulted in an error"` so there's not much point adding additional (but unhelpful) information here.

Don't feel strongly, if we don't have more useful information to add at this point that's fair enough

Reply:
I'd be fine with removing it. Any further information should already be in the report. This is all information we have at this point.

Reply:
Currently, the attachment is misleading, so I removed it again. As I want to revisit cancelation, anyway, I don't really see the point in just adding an attachment for the sake of an attachment. I expect, that the error message already contains better information.